doc/oper.xml

```diff
@@ -1964,3 +1964,157 @@ gap> ModularProduct(NullDigraph(0), CompleteDigraph(10));
+  This is a default function to be passed into the ExecuteDFS function.
+  This does nothing and can be used in place of the PreOrderFunc, PostOrderFunc,
+  AncestorFunc and/or CrossFunc of the ExecuteDFS function.
+  See <Ref Func="ExecuteDFS" Label="for a record, object, integer and 4 functions"/>.
+  <Example><![CDATA[
+gap> PreOrderFunc := function(record, data)
+>       data.num_vertices := data.num_vertices + 1;
+>    end;;
+gap> record := NewDFSRecord(CompleteDigraph(2));;
+gap> data := rec(num_vertices := 0);;
+gap> ExecuteDFS(record, data, 1, PreOrderFunc,
+>               DFSDefault, DFSDefault, DFSDefault);
+gap> data;
+rec( num_vertices := 2 )
+gap> record := NewDFSRecord(CompleteDigraph(2));;
+gap> ExecuteDFS(record, [], 1, DFSDefault,
+>               DFSDefault, DFSDefault, DFSDefault);
+gap> record;
+rec( child := 1, current := 1, 
+  graph := <immutable complete digraph with 2 vertices>, 
+  parent := [ 1, 1 ], postorder := [ 2, 1 ], preorder := [ 1, 2 ], 
+  stop := false )
+]]></Example>
+</Description>
+</ManSection>
+<#/GAPDoc>
+
+<#GAPDoc Label="ExecuteDFS">
+<ManSection>
+  <Oper Name="ExecuteDFS" Arg="record, data, start, PreOrderFunc, PostOrderFunc, AncestorFunc,
+                               CrossFunc"/>
+  <Description>
+  This performs a full depth first search from the <A>start</A> vertex (where <A>start</A> is within the graph).
+  The depth first search can be terminated by changing the <A>record</A>.stop attribute to true in the
+  <A>PreOrderFunc</A>, <A>PostOrderFunc</A>, <A>AncestorFunc</A> or <A>CrossFunc</A> functions.
+  ExecuteDFS takes 7 arguments:
+  <List>
+    <Item><A>record</A>: the depth first search record (created using NewDFSRecord)</Item>
+    <Item><A>data</A>: an object that you want manipulate in the functions passed.</Item>
+    <Item><A>start</A>: the vertex where we begin the depth first search.</Item>
+    <Item><A>PreOrderFunc</A>: this function is called when a vertex is first visited. This vertex
+    is stored in <A>record</A>.current</Item>
+    <Item><A>PostOrderFunc</A>: this function is called when a vertex has no more unvisited children
+    causing us to backtrack. This vertex is stored in <A>record</A>.child and its parent is stored 
+    in <A>record</A>.current</Item>
+    <Item><A>AncestorFunc</A>: this function is called when (<A>record</A>.current,
+    <A>record</A>.child) is an edge and <A>record</A>.child is an ancestor of <A>record</A>.current. An ancestor here means that
+    <A>record</A>.child is on the same branch as <A>record</A>.current but was visited prior to <A>record</A>.current</Item>
+    <Item><A>CrossFunc</A>: this function is called when (<A>record</A>.current,
+    <A>record</A>.child) is an edge and <A>record</A>.child has been visited before <A>record</A>.current
+      and it is not an ancestor of <A>record</A>.current</Item>
+  </List>
+  Note that this function only performs a depth first search on the connected graph that has <A>start</A>.
```

Comment:
```suggestion
  Note that this function only performs a depth first search on the connected component that contains <A>start</A>.
```

Reply:
I think however that this line should be completely changed to "the digraph induced by the vertices reachable from <A>start</A>.
In undirected graphs this would indeed be the connected component containing start. However in directed graphs we might have a connected graph and vertices that are not reachable from the start vertex :)

Reply:
Yes, this is much clearer!